scene/2d/cpu_particles_2d.cpp

```diff
@@ -166,10 +166,20 @@ void CPUParticles2D::_update_mesh_texture() {
 	vertices.push_back(-tex_size * 0.5 + Vector2(tex_size.x, tex_size.y));
 	vertices.push_back(-tex_size * 0.5 + Vector2(0, tex_size.y));
 	PoolVector<Vector2> uvs;
-	uvs.push_back(Vector2(0, 0));
-	uvs.push_back(Vector2(1, 0));
-	uvs.push_back(Vector2(1, 1));
-	uvs.push_back(Vector2(0, 1));
+	AtlasTexture *at = Object::cast_to<AtlasTexture>(*texture);
+	if (!at) {
+		uvs.push_back(Vector2(0, 0));
+		uvs.push_back(Vector2(1, 0));
+		uvs.push_back(Vector2(1, 1));
+		uvs.push_back(Vector2(0, 1));
+	} else {
+		Rect2 region_rect = at->get_region();
+		Size2 atlas_size = at->get_atlas()->get_size();
```

Comment:
For reference, this is where the crash happens. An atlas texture is not guaranteed to have an atlas, so calling `get_size()` on a null reference produces a crash.